src/frequenz/channels/util/_select.py

```diff
@@ -117,82 +80,41 @@ async def stop(self) -> None:
         await asyncio.gather(*self._pending, return_exceptions=True)
         self._pending = set()
 
-    async def ready(self) -> bool:
+    async def ready(self) -> AsyncIterator[Set[Receiver[Any]]]:
```

Comment:
True. I thought of updating the name but I thought it still made sense thinking that it returns a set of ready receivers, so like "get me the ready receivers", but I'm open to rename it.

Reply:
I think `ready` doesn't work so well anymore,  because it is called only once,  where as in the original interface,  it was called after iteration.

Also, if we rename, it would then be a proper break from the previous interface,  instead of having the same function but not working the same way for some reason.